environment/src/main/java/com/sequenceiq/environment/parameters/validation/validators/parameter/AzureParameterValidator.java

```diff
@@ -0,0 +1,67 @@
+package com.sequenceiq.environment.parameters.validation.validators.parameter;
+
+import static com.sequenceiq.environment.parameters.dao.domain.ResourceGroupCreation.USE_EXISTING;
+import static com.sequenceiq.environment.parameters.dao.domain.ResourceGroupUsagePattern.USE_MULTIPLE;
+
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cloud.azure.client.AzureClient;
+import com.sequenceiq.cloudbreak.cloud.azure.client.AzureClientService;
+import com.sequenceiq.cloudbreak.cloud.init.CloudPlatformConnectors;
+import com.sequenceiq.cloudbreak.cloud.model.CloudCredential;
+import com.sequenceiq.cloudbreak.common.mappable.CloudPlatform;
+import com.sequenceiq.cloudbreak.featureswitch.AzureSingleResourceGroupFeatureSwitch;
+import com.sequenceiq.cloudbreak.validation.ValidationResult;
+import com.sequenceiq.cloudbreak.validation.ValidationResult.ValidationResultBuilder;
+import com.sequenceiq.environment.credential.v1.converter.CredentialToCloudCredentialConverter;
+import com.sequenceiq.environment.environment.dto.EnvironmentDto;
+import com.sequenceiq.environment.parameters.dto.AzureParametersDto;
+import com.sequenceiq.environment.parameters.dto.AzureResourceGroupDto;
+import com.sequenceiq.environment.parameters.dto.ParametersDto;
+
+@Component
+public class AzureParameterValidator implements ParameterValidator {
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Inject
+    private AzureClientService azureClientService;
+
+    @Inject
+    private CredentialToCloudCredentialConverter credentialToCloudCredentialConverter;
+
+    @Inject
+    private AzureSingleResourceGroupFeatureSwitch azureSingleResourceGroupFeatureSwitch;
+
+    @Override
+    public ValidationResult validate(EnvironmentDto environmentDto, ParametersDto parametersDto, ValidationResultBuilder validationResultBuilder) {
+        AzureParametersDto azureParametersDto = parametersDto.azureParametersDto();
+        if (!azureSingleResourceGroupFeatureSwitch.isActive() || Objects.isNull(azureParametersDto)) {
+            return validationResultBuilder.build();
+        }
+
+        AzureResourceGroupDto azureResourceGroupDto = azureParametersDto.getAzureResourceGroupDto();
+        if (Objects.isNull(azureResourceGroupDto)
+                || USE_MULTIPLE.equals(azureResourceGroupDto.getResourceGroupUsagePattern())
+                || !USE_EXISTING.equals(azureResourceGroupDto.getResourceGroupCreation())) {
+            return validationResultBuilder.build();
```

Comment:
We should use ValidationResult.empty() instead

Reply:
Since the validationResultBuilder is passed in as a parameter, we cannot be sure that there is no validation result in it. Currently it would be ok to use as you suggest, but later on if a validation is added before the invocation of the method I think a developer can easily forget about downstream used ValidationResult.empty usages.
What do you think?

Reply:
Ok